receiver/prometheusexec/subprocessmanager/manager.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package subprocessmanager
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"math"
+	"math/rand"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/kballard/go-shellquote"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/prometheusexec/subprocessmanager/config"
+)
+
+// Process struct holds all the info needed for subprocesses
+type Process struct {
+	Command    string
+	Port       int
+	Env        []config.EnvConfig
+	CustomName string
+}
+
+const (
+	// HealthyProcessTime is the default time a process needs to stay alive to be considered healthy
+	HealthyProcessTime time.Duration = 30 * time.Minute
+	// healthyCrashCount is the amount of times a process can crash (within the healthyProcessTime) before being considered unstable - it may be trying to find a port
+	HealthyCrashCount int = 3
+	// delayMutiplier is the factor by which the delay scales (default is doubling every crash)
```

Comment:
```suggestion
	// delayMutiplier is the factor by which the delay scales
```
(the default is implied by the value since this is a `const`)

Reply:
Will change, thanks!